api/v1/storagecluster_types.go

```diff
@@ -137,11 +140,41 @@ type ManageCephObjectStoreUsers struct {
 	ReconcileStrategy string `json:"reconcileStrategy,omitempty"`
 }
 
+// ExternalStorageKind specifies a kind of the external storage
+type ExternalStorageKind string
+
+const (
+	// KindOcs specifies a "ocs" kind of the external storage
+	KindOcs ExternalStorageKind = "ocs"
+
+	// KindRhcs specifies a "rhcs" kind of the external storage
+	KindRhcs ExternalStorageKind = "rhcs"
+)
+
 // ExternalStorageClusterSpec defines the spec of the external Storage Cluster
-// to be connected to the local cluster
+// to be connected to the external cluster
 type ExternalStorageClusterSpec struct {
-	// +optional
+	//+kubebuilder:default:=true
+	// Enable enables the external mode
 	Enable bool `json:"enable,omitempty"`
+
+	//+kubebuilder:default:=rhcs
+	// StorageProviderKind describes the kind of the external storage
+	StorageProviderKind ExternalStorageKind `json:"storageProviderKind,omitempty"`
+
+	// ConnectionString describes the blob of provider
+	ConnectionString string `json:"connectionString,omitempty"`
+
+	//+kubebuilder:default:="1Ti"
```

Comment:
I'm not in favour of setting this default here. Let the StorageCluster have the requested value set explicitly.

Reply:
@nbalacha This is not an initial value but needs to be set from the start, with a specific setting we will not be able to onboard. 
The problem is that in order to support backword compatibility with the external mode we need to be able to not specify this field which bring us to the default value solution

Reply:
Also, to the best of my understanding,  the default value should not affect the value in the CT itself. Only the in-memory struct created when reading the CR and in that case we can just ignore it in RHCS external mode

Reply:
But isn't the deployer creating the storagecluster CR and getting the capacity via the UI? Why then should this be a default value? It should be set in the StorageCluster CR created by the deployer.

Reply:
This is a product feature and as such, we need to treat this as an external API, deployer, or no deployer. 
So we can either choose to make the field non-optional, which is problematic because we need to support backword compatibility, or provide a default value and ignore it in the case of RHCS.

Reply:
I'm not sure I understand. If you make the field optional, and require it to be set for OCS, won't it work?

Reply:
How would you suggest doing that? a reconcile failure? that is an anti-pattern as this is a CR configuration failure. 
I am aware of only 2 ways to deal with this:
1. validation webhook - out of scope right now
2. default value - no apparent downsides as we are already going to touch the external mode code inside ocs operator.

Reply:
We also do not provide default values for OSDs in the CRD. I do not see why the same logic does not hold for the external mode in the managed service.

Reply:
Think about that differently - the default value allows us to broadcast the actual default value as part of the API itself. 
Assuming we will not take that route we will have to either treat that as a failure (my prev response) or provide a default value hardcoded in golang - which I think is a worse version of the current solution as every piece of code that needs the actual value will have to implement the same logic which can lead to bugs.

We also do not provide default values for OSDs in the CRD - when that was implemented we used the tools that were available for us then. Now we have new tools

Reply:
I don't like the idea of having a value as part of the API. I would prefer that the value be set in the CR explicitly.
However, I would like to hear from Jose and Umanga on this point.

Reply:
Ok, but just another point, if we do not publish the default value in the API it means that we might (by mistake or intention) change it in future code changes which is problematic for such kind of value

Reply:
I am not in favour of having the default value in the code at all. In my opinion, the best way would be to explicitly specify the value in the CR and check in the reconcile loop. This PR already has to check to validate that the connection string has been provided. This would be an additional check.

Reply:
@nbalacha We cannot do that as we cannot make this field mandatory for backward compatibility sake! 
If you have a way to deal with that please explain.

Just to clarify, I do agree that if this was a new section we would mark this field as mandatory and force the client who creates the CR to provide a value.

Reply:
Run time checks and reconcile failures are the worst solution as it has no guarantees around what was acted upon and what not and is an anti-pattern for static checks

Reply:
> It should be optional and the check-in the code would be to ensure that the field value is provided if the kind is OCS.
This is the exact case I am advocating against.

Reply:
Since it's a new field, it's ok to set sensible default. We just need to be careful about how we consume it and have necessary validations.

Reply:
After discussing it offline we came to a conclusion that we should remove the default tag, so removing it.